modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -1036,19 +1037,34 @@ public String toEnumVarName(String value, String datatype) {
 
     @Override
     public String toEnumValue(String value, String datatype) {
-        if ("Integer".equals(datatype) || "Double".equals(datatype)) {
+        String sanitzedDataType = sanitizeForEnum(datatype);
```

Comment:
Who is calling this method?

I did not check it yet, I think we could add a parameter and provide the mostInnerItems type (will be added with #66)

Reply:
The method is called into updateCodegenPropertyEnum.
I've done a second try where I search the most inner datatype on this method, so I think you're right, I can replace my call to findMostInnerDatatype method with a simple var.mostInnerItems.datatype with #66 .